controller/gateway/controller_watch.go

```diff
@@ -17,6 +17,7 @@ import (
 	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	operatorv1beta1 "github.com/kong/kubernetes-configuration/v2/api/gateway-operator/v1beta1"
+	operatorv2alpha1 "github.com/kong/kubernetes-configuration/v2/api/gateway-operator/v2alpha1"
```

Comment:
WDYT about using https://github.com/Kong/kong-operator/blob/57307f0d56c4c5e109baab66696d3b957c1ab5a5/controller/gateway/types.go#L15 (or https://github.com/Kong/kong-operator/blob/0356fc46538454b8d856662de5f979d1b92d6abd/internal/types/operatortypes.go#L46 if you prefer, the former is an alias to this).

This is already in place to drive the `GatewayConfiguration` used in this repo. WDYT?

Whatever's our decision on this I believe we should be consistent, so that we either use it or drop these alias types entirely.

Reply:
I do see your point but OTOH changes like the one in this PR would require only 1 line to be changed.

Do you feel that chasing all the references of a package that's being changed is worth throwing away all the aliases?

Reply:
sure, let's discuss it in the team sync

Reply:
Here's a PR that takes care of enforcing the `ControlPlane` related types (incomplete yet but a good starter).

I'll close this discussion as we do not want to handle that in you PR here.